osbuild/formats_common.py

```diff
@@ -0,0 +1,45 @@
+"""
+Common procedures and classes between the v1 and v2 formats
+"""
+
+from abc import abstractmethod
+
+
+class OSbuildError(Exception):
+    """
+    All exceptions that want to be formated in json for a user program must overload OSbuildError.
+    They also must be declared in the map 'map_error_id'.
+    """
+
+    @abstractmethod
+    def payload_as_dict(self):
+        pass
+
+    @property
+    def ID(self):
+        return OSbuildError.map_error_id.get(self.__class__, 0)
+
+    map_error_id = {}
+
+    @staticmethod
+    def add_error_id(error_klass):
+        OSbuildError.map_error_id[error_klass.__class__] = len(OSbuildError.map_error_id)+1
+
+
+# pylint: disable=inconsistent-return-statements #Given an exception, there always be something to catch and return
+# pylint: disable=broad-except #We need to catch all exceptions here to have a default return value
+def format_exception(error: Exception):
+    """produces a json output from an OSbuildError"""
+    try:
+        raise error
+    except OSbuildError as e:
+        return {
+            "id": e.ID,
+            "human-readable": f"{e}",
+            "payload": e.payload_as_dict()
```

Comment:
For the actual formatting: https://api.openshift.com, already defines an error schema, we might as well use that, i.e.

kind | stringIndicates the type of this object. Will always be 'Error'
-- | --
id | integer($int32)Numeric identifier of the error.
href | stringSelf link.
code | stringGlobally unique code of the error, composed of the unique identifier of the API and the numeric identifier of the error. For example, for if the numeric identifier of the error is 93 and the identifier of the API is clusters_mgmt then the code will be CLUSTERS-MGMT-93.
reason | stringHuman readable description of the error.
details | {description:Extra information about the error.} | description: | Extra information about the error.
description: | Extra information about the error.

The question is if we should wrap this or just expose as is, and thus use `kind` as our type designator for v2 results. Alternatively, we could keep our current `result` key and than wrap it, like:
```
{
   "type": "error",
   "error": {
      "kind": "error",
      "code": ...
   }
}
```

Reply:
good idea.

Reply:
I went for:
```
{
            "type": "error",
            "error": {
                    "id": self.ID,
                    "code": self.code,
                    "reason": str(self),
                    "details": self.details
            }
        }

```